src/openms/include/OpenMS/ANALYSIS/OPENSWATH/OpenSwathOSWWriter.h

```diff
@@ -103,6 +104,7 @@ namespace OpenMS
                        const String& input_filename = "inputfile",
                        bool ms1_scores = false,
                        bool sonar = false,
+                       bool compute_peak_shape = false,
```

Comment:
the interface is very easy to use wrong (in general), since its just a bunch of bools -- easy to get the order wrong.
In this particular case its even more dangerous, since old code with
```
OpenSwathWriter("bla", 0, "myfile", true, false, true)
```
still works after this change, but does something completely different!

Suggestions:
1) replace each boolean with an ENUM, e.g. `enum class SONAR {ON, OFF}` which results in `..., SONAR:ON, ...` etc

Use a struct/class which contains the values and allows to set proper names via chaining
```
myParams().sonar(false).computePeakShape(true). ...
```

Reply:
would chaining work with pyopenms?

Reply:
not a pyOpenMS expert, but each functions simply returns a reference to self.
We should not return a copy here, since the changes will be lost unless assigned to a final result object, i.e.
`OpenSwathWriter("bla", 0, "myfile", myParams().sonar(false).computePeakShape(true))` only works when `MyParam& sonar(bool state)` returns a reference, not a copy.

Reply:
@jpfeuffer I always get confused with how temporaries are handled in pyOpenMS. Would in principle something like @cbielow proposed work?

Reply:
I think the default wrapping for pyopenms is to always make a copy. You will need a custom pyx or a new wrap-statement to achieve that. (e.g. wrap-return-self where you do something like

```
self.inst.get().function()
return self
```
)

Reply:
Would be kinda nice indeed

Reply:
hmm better but using the enum is here also a bit non-standard.
Without looking at the details, could another option be to use the OpenMS DefaultParamHandler?
e.g. similar to here: https://github.com/OpenMS/OpenMS/blob/57817d632cf70d45eeb52817c3f6afc2f54921f4/src/openms/source/FORMAT/MSPFile.cpp

Reply:
We could however I am not sure if this is a bit overkill for this application? Theoretically, none of these parameters should be set manually they should all be read off of the parameters set with OpenSwathWorkflow? I am not sure if there is a way for the class to just determine its own parameters based on the feature it is provided? E.g. look for SONAR data and if it is found than initiate with SONAR?

Reply:
So it is exposed to python but automatic handling might still work? I am going to look into it further